### ISSUE TITLE
TS-CRF

### ISSUE THREAD

**Issue by GaoQiyu** [chrischoy/SpatioTemporalSegmentation#10]
I trained ResUNetIN14 with Adam(lr=0.1,weight_decay=1e-4,) on S3DIS fold1 split
but it only achieve 30%mIOU......
Then I test the trained model and found points on the same object have multiple labels
I think maybe the reason is no CRF?

**Comment by chrischoy** [comment]
Repost the same response again:

Instance Norm is worse than BN. That is why I removed all IN instantiation. I added only for experiments. Also, SGD is better than Adam if you use BN. CRF for static scenes don't really help much.

What batch_size are you using? If you can't squeeze in a lot of batches, try increasing `iter_size`. This is effectively gradient accumulation.

Also, it is important to set the collision labels as ignore_label OR use higher resolution.

**Comment by GaoQiyu** [comment]
I use batch_size = 8
sorry I can't find how to set collision labels
and ...
I've applied many times Scannet dataset， but no reply....
Do you know how to get this dataset..??

**Comment by chrischoy** [comment]
There's only one way to get the scannet through the official channel.

30% is really low. How long have you trained it for? what was the voxel size?

Sorry, the collision label was done with the GPU voxelizer which we removed for the final version. However, the performance drop is insignificant for the ScanNet.

**Comment by GaoQiyu** [comment]
with voxel size 2cm 
trained 50 epoch on S3DIS(30epoch acheieve 30%, 50epoch still 30%)
do you mean if I use voxelizer in this code then I don't need to set collision label
or just I don't need to set collision label?
30% is really....really piss me off.

**Comment by chrischoy** [comment]
Haha, Stanford dataset is extremely small

30 epoch on the stanford dataset is 1k iterations with batch_size 8. However, 50 epoch 30% seems quite problematic.

I'm currently under a bit of load, but let me get back to you with the Stanford training scripts.

**Comment by GaoQiyu** [comment]
very thanks!

**Comment by chrischoy** [comment]
This is a pretty old log that I found. I could not find one for ResUNet18 which is what you are using, but the trend should look similar.

The score is the Overall Accuracy.

0-15k: https://pastebin.com/X1y7TSvR
15k-30k: https://pastebin.com/57i4rRD5
30k-45k: https://pastebin.com/22vgGMev
45k-60k: https://pastebin.com/F5wkyQcD

**Comment by GaoQiyu** [comment]
I am thinking whether it's S3DIS data's problem.
because it looks like too dirty compared with Scannet
have you try to train model only on S3DIS before?

**Comment by chrischoy** [comment]
I've never trained the S3DIS with scannet before. Look at the log.

Are you using the provided data augmentation at all?

**Comment by GaoQiyu** [comment]
I mean have you trained Res16UNet34C or other pointcloud segmentation model using minkowskii engine only on S3DIS fold1?
Yes I've follow your code including dataset and voxelizer part, and set the parameters according to the log.

If it's not the problem of dataset...I will check my code again

**Comment by chrischoy** [comment]
The Stanford training script has been added in the commit https://github.com/chrischoy/SpatioTemporalSegmentation/commit/75c33f56b3d00a80a43df03ab0fa4f5ef8f226ad
